cloud/scope/clients.go

```diff
@@ -79,46 +63,14 @@ func (c *AzureClients) setCredentials(subscriptionID string) error {
 		return err
 	}
 
-	if subscriptionID == "" {
-		subscriptionID = settings.GetSubscriptionID()
-		if subscriptionID == "" {
-			return fmt.Errorf("error creating azure services. subscriptionID is not set in cluster or AZURE_SUBSCRIPTION_ID env var")
-		}
-	}
-
-	c.subscriptionID = subscriptionID
-	c.tenantID = strings.TrimSuffix(settings.Values[auth.TenantID], "\n")
-	c.clientID = strings.TrimSuffix(settings.Values[auth.ClientID], "\n")
-	c.clientSecret = strings.TrimSuffix(settings.Values[auth.ClientSecret], "\n")
-
-	c.environment = settings.Values[auth.EnvironmentName]
-	if c.environment == "" {
-		c.environment = azure.PublicCloud.Name
-	}
-
+	c.EnvironmentSettings = settings
 	c.ResourceManagerEndpoint = settings.Environment.ResourceManagerEndpoint
-	c.ResourceManagerVMDNSSuffix = GetAzureDNSZoneForEnvironment(settings.Environment.Name)
-	settings.Values[auth.SubscriptionID] = subscriptionID
-	settings.Values[auth.TenantID] = c.tenantID
+	c.ResourceManagerVMDNSSuffix = settings.Environment.ResourceManagerVMDNSSuffix
```

Comment:
is there a unit test that can be added to make sure that the values get set properly with other clouds? unfortunately we don't have e2e testing in sovereign clouds yet so a unit test can help make sure this doesn't cause any regressions for other clouds

Reply:
@CecileRobertMichon isn't this already that test? https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/master/cloud/scope/clients_test.go#L36

Reply:
indeed 💯